test/fw/bin/pbs_snapshot

```diff
@@ -0,0 +1,193 @@
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software - under
+# a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+import sys
+import os
+import getopt
+import tempfile
+import time
+import logging
+import errno
+import signal
+
+import ptl
+from ptl.utils.pbs_cliutils import CliUtils
+from ptl.lib.pbs_testlib import *
+from ptl.utils.pbs_snaputils import *
+from ptl.utils.pbs_dshutils import DshUtils
+
+
+def trap_exceptions(etype, value, tb):
+    """
+    Trap SIGINT and SIGPIPE
+    """
+    # This is done so that any exceptions created by this method itself
+    # are caught by the default excepthook to prevent endless recursion
+    sys.excepthook = sys.__excepthook__
+
+    if issubclass(etype, KeyboardInterrupt):
+        pass
+    elif issubclass(etype, IOError) and value.errno == errno.EPIPE:
+        pass
+    else:
+        sys.__excepthook__(etype, value, tb)
+
+    # Set syshook back to trap_exceptions to catch future exceptions
+    sys.excepthook = trap_exceptions
+
+sys.excepthook = trap_exceptions
+
+
+def usage():
+    msg = """
+Usage: pbs_snapshot -o <path to output tar file> [OPTION]
+
+    Take snapshot of a PBS system and optionally capture logs for diagnostics
+
+    -d <pbs_diag>                     diag directory to use as input
+    -H <hostname>                     hostname to operate on
+                                      Defaults to value of PBS_SERVER
+    -l <loglevel>                     set log level to one of INFO, INFOCLI,
+                                      INFOCLI2, DEBUG, DEBUG2, WARNING, ERROR
+                                      or FATAL
+    --service-logs=<num days>         number of service logs to collect
+    --accounting-logs=<num days>      number of accounting logs to collect
+    --additional_hosts=<hostname>     collect data from additional hosts
+                                      'hostname' is a comma separated list
+    --map=<file>                      file to store the map of obfuscated data
+    --obfuscate                       obfuscates sensitive data
+    --version                         print version number and exit
+"""
+
+    print msg
+
+
+if __name__ == '__main__':
+
+    # Arguments to PBSSnapUtils
+    out_dir = None
+    diag_path = None
+    server_host = None
+    log_level = "INFOCLI2"
+    acct_logs = 30  # Capture 30 days of accounting logs by default
+    srvc_logs = 5   # Capture 5 days of service logs by default
+    additional_hosts = None
+    map_file = None
+    anonymize = False
+    create_tar = True
+
+    log_file = "pbs_snapshot.log"
+
+    PtlConfig()
+
+    # Parse the options provided to pbs_snapshot
+    try:
+        sopt = "d:H:l:o:h"
```

Comment:
we missed it but -o was actually there already, I've added the usage for -h, thanks